language/move-lang/src/shared/mod.rs

```diff
@@ -227,3 +269,84 @@ pub fn format_delim<T: fmt::Display, I: IntoIterator<Item = T>>(items: I, delim:
 pub fn format_comma<T: fmt::Display, I: IntoIterator<Item = T>>(items: I) -> String {
     format_delim(items, ", ")
 }
+
+//**************************************************************************************************
+// Flags
+//**************************************************************************************************
+
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub enum Flag {
+    Test,
+}
```

Comment:
Maybe if you do some derive_struct opt here or something it might work out so you don't need a custom one